### DISCUSSION TITLE
Extra Intelliflow Pumps showing in dashpanel that I cannot delete

### DISCUSSION THREAD

**Discussion by loopy321** [tagyoureit/nodejs-poolController#423] [Q&A] [answered]
After updating from v. 6, I see extra pumps showing up that were not there before (I only have one intelliflow) and I cannot delete them via the config.  Any ideas how to troubleshoot/fix?  (See below, and I have attached a replay of the deleting which does not work).  Thanks!! 
![image](https://user-images.githubusercontent.com/1026345/164720923-e7c33081-6f9b-4daa-8c70-8f1e92da347b.png)
[replay (pump deletes).zip](https://github.com/tagyoureit/nodejs-poolController/files/8541288/replay.pump.deletes.zip)

**Comment by rstrouse** [comment]
I'll have a look.  I have seen this with IntelliTouch where the panel was reporting all 8 slots but I haven't seen it in EasyTouch.  They turned something off on the panel and it went away.  When 8 pumps are involved, there is waaaaay more traffic on the RS485 bus.  If we can make EasyTouch stop reporting these, your system may feel more responsive.

**Reply by loopy321** [reply]
Thanks!  Not sure if it helps, but prior to the update, I believe it only showed the one pump.

**Reply by rstrouse** [reply]
Thanks.  Good info to know.

**Reply by loopy321** [reply]
In case it helps I did a grep of my logfile for "pump" which includes prior to the upgrade change on 2022-03-21, but I could not figure out what happened differently.  See attached.
[Log-grep-pump_across-update-on-2022-03-21.txt](https://github.com/tagyoureit/nodejs-poolController/files/8541844/Log-grep-pump_across-update-on-2022-03-21.txt)

**Reply by rstrouse** [reply]
I figured it out.  It turns out that older firmwares assumed that the pump types were populated with 255 for all 16 slots.  Even though EasyTouch only allows 2 pumps it still reports 16 pumps but marks the type as 255.  But the older firmware reports 255 as a VS3050 pump.  I will have a fix shortly as the code just needs to detect the condition where the pump number is greater than the max available pumps.

**Comment by rstrouse** [comment, marked as answer]
Go ahead and pull njsPC.  It may clean them up on its own or you may need to go in and delete them.

**Reply by loopy321** [reply]
Pulled and restarted, pumps still there, but deleted the extra pumps and hard refresh on dashpanel.  Now all is well.  
Awesome!  Thank you for the help.
